File: VagasOnlineProjetoFE/vagasonlinefrontend/src/services/vagasService.js
import api from "./api";

export async function getVagas() {
  const response = await api.get("/vagas");
  return response.data;
}

export async function getVagaById(id) {
  const response = await api.get(`/vagas/${id}`);
  return response.data;
}

export async function createVaga(vaga) {
  const response = await api.post("/vagas", vaga);
  return response.data;
}

export async function updateVaga(id, vaga) {
  const response = await api.put(`/vagas/${id}`, vaga);
  return response.data;
}

export async function deleteVaga(id) {
  const response = await api.delete(`/vagas/${id}`);
  return response.data;
}

export async function registrarInteresse(interesse) {
  const response = await api.post("/vagas/interesse", interesse);
  return response.data;
}

File: Android/app/src/main/java/com/example/appandroid/api/VagasService.java
package com.example.appandroid.api;

import com.example.appandroid.model.Interesse;
import com.example.appandroid.model.Vaga;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface VagasService {
    @GET("vagas")
    Call<List<Vaga>> getVagas();

    @POST("vagas/interesse")
    Call<Void> registrarInteresse(@Body Interesse interesse);
}